gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/ExecutionContext.java

```diff
@@ -215,6 +216,36 @@ public void removePendingSelectionSourceForOrderBy(String source) {
     pendingSelectionSourcesForOrderBy.remove(source);
   }
 
+  public void removeSelectionAttributes(String source, Set<String> attributes) {
+    if (!sourceToSelectionExpressionMap.containsKey(source)) {
+      return;
+    }
+
+    // converting to a mutable map
+    Map<String, List<Expression>> selectionExpressionMap =
```

Comment:
nit: guava lib has some fancy Maps.filterEntries, Maps.filterValues etc methods. See if they can be helpful here.